tests/rados/test_reads_balancer.py

```diff
@@ -278,8 +282,175 @@ def run(ceph_cluster, **kw):
+                        )
+                    except Exception as e:
+                        log.error(
+                            f"Exception hit during setting the read recommendations. {e}"
+                        )
+                        log.debug(
+                            "The exception hit is expected due to bug :"
+                            " https://bugzilla.redhat.com/show_bug.cgi?id=2237574."
+                            " The below exception to be uncommented post bug fix."
+                        )
+                        # raise Exception("read recommendations not set")
+
+                    log.info(
+                        f"Successfully set the read recommendations on the pool : {pool_name}"
+                    )
+                    time.sleep(5)
+                    # There should be no data movement b/w the PGs when read balancing was performed.
+                    # Checking the same if we had clean pgs to start with
+                    if clean_pgs:
+                        if not rados_obj.check_pool_pg_states(
+                            pool=pool_name, disallowed_states=disallowed_states
+                        ):
+                            log.error(
+                                "All the PGs are not active + clean post addition of reads balancing"
+                                f"There should be no data movement. Observed with pool {pool_name}"
+                            )
+                            raise Exception("Data movement post reads balancing error")
+                        log.debug(f"Clean PGs on pool {pool_name} post reads balancing")
+
+                    log.debug("Sleeping for 20 seconds for scores to be updated")
+                    time.sleep(20)
+                    pool_details = rados_obj.get_pool_details(pool=pool_name)
+                    final_read_scores = pool_details["read_balance"]
+                    final_score = init_read_scores["score_acting"]
+                    log.info(
+                        f"After applying read balancing on pool {pool_name} the Read score is : {final_score}\n"
+                        f"Final scores Fetched from pool is {final_read_scores}"
+                    )
+
+                    # The final scores should be less than or equal to initial scores
+                    if init_score < final_score:
+                        log.error(
+                            f"The scores are worse post application of reads balancing for pool {pool_name}"
+                            f"init score {init_score} ---- Final score : {final_score}"
+                        )
+                        raise Exception("Final scores are not correct")
+                    log.info(
+                        f"Completed offline tool scenario on pool {pool_name} for reads balancing"
+                    )
+
+                    # tbd: Checking the read throughput of the pools post reads balancer application
```

Comment:
let's break here and put the EC pool workflow in separate for loop just like we did for online command verification test, this way we would just be running the test for one replicated pool and one ec pool instead of all

Reply:
Unless the intention is to perform the workflow for all the pools

Reply:
Intention is to perform this on all the pools.